MapboxGeocoderTests/BatchForwardGeocodingTests.swift

```diff
@@ -0,0 +1,145 @@
+        options.allowedISOCountryCodes = ["CA"]
+
+        let task = geocoder.batchGeocode(options) { (placemarks, attribution, error) in
+            let results = placemarks![0]
+            let attribution = attribution![0]
+        
+            XCTAssertEqual(results.count, 5, "single forward batch geocode should have 5 results")
+            
+            XCTAssertEqual(attribution, "© 2017 Mapbox and its suppliers. All rights reserved. Use of this data is subject to the Mapbox Terms of Service. (https://www.mapbox.com/about/maps/)")
+            
+            expectation.fulfill()
+        }
+        XCTAssertNotNil(task)
+        
+        waitForExpectations(timeout: 1) { (error) in
+            XCTAssertNil(error, "Error: \(error!)")
+            XCTAssertEqual(task.state, URLSessionTask.State.completed)
+        }
+    }
+    
+    func testValidForwardMultipleBatchGeocode() {
+        let expectation = self.expectation(description: "forward batch geocode with multiple queries should return results")
+        
+        _ = stub(condition: isHost("api.mapbox.com")
+            && isPath("/geocoding/v5/mapbox.places-permanent/20001;20001;20001.json")
+            && containsQueryParams(["country": "us", "access_token": BogusToken])) { _ in
+                let path = Bundle(for: type(of: self)).path(forResource: "permanent_forward_multiple_valid", ofType: "json")
+                return OHHTTPStubsResponse(fileAtPath: path!, statusCode: 200, headers: ["Content-Type": "application/vnd.geo+json"])
+        }
+        
+        let geocoder = Geocoder(accessToken: BogusToken)
+        
+        let options = ForwardBatchGeocodeOptions(queries: ["20001", "20001", "20001"])
+        options.allowedISOCountryCodes = ["US"]
+        
+        let task = geocoder.batchGeocode(options) { (placemarks, attribution, error) in
+            
+            let queries = placemarks!
+            let attribution = attribution![0]
+            
+            XCTAssertEqual(queries.count, 3, "forward batch geocode should have 3 queries")
+            
+            for result in queries {
+                XCTAssertEqual(result.count, 5, "each forward batch geocode query should have 5 found results")
+                XCTAssertEqual(attribution, "© 2017 Mapbox and its suppliers. All rights reserved. Use of this data is subject to the Mapbox Terms of Service. (https://www.mapbox.com/about/maps/)")
+            }
+            
+            let sampleResult = queries[0][0]
+            
+            XCTAssertEqual(sampleResult.qualifiedName, "Washington, District of Columbia 20001, United States")
+
+            expectation.fulfill()
+        }
+        XCTAssertNotNil(task)
+        
+        waitForExpectations(timeout: 1) { (error) in
+            XCTAssertNil(error, "Error: \(error!)")
+            XCTAssertEqual(task.state, URLSessionTask.State.completed)
+        }
+    }
+    
+    func testInvalidForwardSingleBatchGeocode() {
```

Comment:
@1ec5 both `testInvalidForwardSingleBatchGeocode` / `testInvalidForwardMultipleBatchGeocode ` tests are commented out because they are failing. I'm actually seeing the same error the original issue described - `Expected to decode Array<Any> but found a dictionary instead.` 🤔

Reply:
Which decode step is failing? If the step that tries to look for a `message` in a dictionary in an array (`[GeocoderAPIResult]`) is failing, then we need to make sure the fallback that looks for `message` in a dictionary (`GeocoderAPIResult`) is working. If the step that tries to decode an array of individual results (`[GeocoderResult]`) is failing, then we should make sure the fallback that looks for a bare result (`GeocoderResult`) is working.